src/Orchard.Web/Modules/Orchard.MediaLibrary.WebSearch/Views/WebSearch/Index.cshtml

```diff
@@ -1,90 +1,102 @@
 ﻿@model Orchard.MediaLibrary.ViewModels.ImportMediaViewModel
 
-@using System.Text
-@using Orchard.ContentManagement
-@using Orchard.MediaLibrary.Models
+@using Orchard.MediaLibrary.WebSearch.Providers
+@using System.Linq
 
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
 <html xmlns="http://www.w3.org/1999/xhtml">
 <head>
     @{
         Script.Require("jQuery").AtFoot();
         Script.Require("Knockout").AtFoot();
-        
+
         Style.Include("orchard-mediaproviders-admin.css");
         Style.Include("orchard-websearch-admin.css");
         Style.Require("FontAwesome");
+
+        var webSearchProviders = WorkContext.Resolve<IEnumerable<IWebSearchProvider>>();
+
+        var hasValidApiKey = webSearchProviders.Any(provider => provider.IsValid);
+        var apiNames = webSearchProviders.Select(provider => provider.Name);
     }
 
     @Display.Metas()
     @Display.HeadScripts()
     @Display.HeadLinks()
     @Display.StyleSheetLinks()
-    
-    @{
-        var apiKey = WorkContext.CurrentSite.As<WebSearchSettingsPart>().ApiKey;
-    }
 </head>
-<body >
+<body>
     <div id="websearch-main">
-        <!-- Register for an API account on https://datamarket.azure.com/dataset/5BA839F1-12CE-4CCE-BF57-A49D98D29A44 -->
-        
-        @if (String.IsNullOrWhiteSpace(apiKey)) {
+        @if (!hasValidApiKey)
+        {
             <div id="message-api">
                 @T("You need to define an API Key in the Media settings before being able to use this feature.")
```

Comment:
Please update this according to new functionality.

Reply:
I'm not sure which new functionality are you referring to. I made another warning message if there are no Web Search providers enabled.

Reply:
Yep, that's what I was referring to!